src/OpenSpaceToolkit/Astrodynamics/Flight/Maneuver.cpp

```diff
@@ -191,7 +219,54 @@ void Maneuver::print(std::ostream& anOutputStream, bool displayDecorator) const
     displayDecorator ? ostk::core::utils::Print::Footer(anOutputStream) : void();
 }
 
-Maneuver Maneuver::ConstantMassFlowRateProfile(
+Maneuver Maneuver::FromTabulatedDynamics(const Shared<Dynamics>& aTabulatedDynamicsSPtr)
+{
+    if (aTabulatedDynamicsSPtr == nullptr || !aTabulatedDynamicsSPtr->isDefined())
+    {
+        throw ostk::core::error::runtime::Undefined("Tabulated Dynamics");
+    }
+
+    // Downcast to TabulatedDynamics
+    Shared<TabulatedDynamics> tabulatedDynamicsSPtr =
+        std::dynamic_pointer_cast<TabulatedDynamics>(aTabulatedDynamicsSPtr);
+
+    if (tabulatedDynamicsSPtr)
+    {
+        const Array<Shared<const CoordinateSubset>> expectedWriteCoordinateSubsets = {
+            CartesianVelocity::Default(), CoordinateSubset::Mass()
+        };
+
+        if (tabulatedDynamicsSPtr->getWriteCoordinateSubsets() != expectedWriteCoordinateSubsets)
+        {
+            throw ostk::core::error::runtime::Wrong("Tabulated Dynamics Write Coordinate Subsets");
+        }
+
+        const MatrixXd contributionProfile = tabulatedDynamicsSPtr->getContributionProfile();
+        Array<Vector3d> accelerationProfile = Array<Vector3d>::Empty();
+        Array<Real> massFlowRateProfile = Array<Real>::Empty();
+
+        for (Size i = 0; i < tabulatedDynamicsSPtr->getInstants().getSize(); i++)
+        {
+            accelerationProfile.add(
+                Vector3d(contributionProfile(i, 0), contributionProfile(i, 1), contributionProfile(i, 2))
```

Comment:
@phc1990 Interesting, thanks for the very thorough comments.

Addressing the comments specifically about my code, I don't think my code makes any assumptions as it does perform the checks you mention in like 239 and throws an error if that coordinate subset order is violated or not present. But yes, I agree that it should be made compatible with `TabulatedDynamics` that have coordinate subsets of Velocity and Mass that are in any order.

I actually prefer your solution #2 the best, and I don't think it would be fairly quick to implement. Regarding option 3, I actually really don't like it personally because even though I see the beauty and reuse of code that it enables, I think it tries to mix too many different concepts into one and could become quite confusing and lead to errors by users in the future. I think that state and dynamics should be kept seperate, because even though they may have the same dimensions and represent similar (or in some cases the same physical quantities), they are used in very different ways by our propagator and numerical integrator, stored in different way (or not stored at all which is indeed a whole other discussion), etc.

However, +1 for actually reusing the concept of expand and reduce for the tabulated dynamics (which is essentially what you are suggestion with option 2) since that is the only Dynamics who's coordinate subsets are not fixed in place or order.

Down to here everyone else's thoughts.

Reply:
@apaletta3 on the code itself:

> I don't think my code makes any assumptions as it does perform the checks you mention in like 239 and throws an error if that coordinate subset order is violated or not present

```cpp
const Array<Shared<const CoordinateSubset>> expectedWriteCoordinateSubsets = {
    CartesianVelocity::Default(), CoordinateSubset::Mass()
};

if (tabulatedDynamicsSPtr->getWriteCoordinateSubsets() != expectedWriteCoordinateSubsets)
{
    throw ostk::core::error::runtime::Wrong("Tabulated Dynamics Write Coordinate Subsets");
}
```

Ok, I see that the assumption about subsets presence and sorting is safe after these line.

**IF** I understood it correctly (I might have missed something 🙃), I still I think the code should just retrieve `CartesianVelocity` and `Mass` in a generic way. Otherwise there are families of `TabulatedDynamics` that should work (✅) when calling `FromTabulatedDynamics` but that wouldn't (❌):

| Subsets   |      Should work      |  Would work |
|----------|:-------------:|:------:|
| `{CartesianVelocity, Mass}` |  ✅  | ✅  |
| `{Mass, CartesianVelocity }` | ✅  | ❌  |
| `{CartesianVelocity, Mass, AnotherSubset}` | ✅  | ❌  |
| `{Mass, CartesianVelocity, AnotherSubset}` | ✅  | ❌  |
| `{CartesianVelocity, AnotherSubset, Mass}` | ✅  | ❌  |
| etc. | ✅  | ❌  |

Although some of these ☝️ are mostly "theoretical", I think that the fact that `{CartesianVelocity, Mass}` would work but `{Mass, CartesianVelocity}` wouldn't might indicate that we are unnecessarily coupled with the `TabulatedDynamics` subsets order. Since this is being coded once, we might just to it generic enough (whilst still correct).

> But yes, I agree that it should be made compatible with TabulatedDynamics that have coordinate subsets of Velocity and Mass that are in any order.

LOL ok we are on the same page, read too quickly 👍 

----

On the rest, I don't have a preference as of today. Even if we leave it as is, I think it's ok (just so that the MR does not blow up). 

Just one comment, one this:

> Regarding option 3, I actually really don't like it personally because even though I see the beauty and reuse of code that it enables, I think it tries to mix too many different concepts into one and could become quite confusing and lead to errors by users in the future. I think that state and dynamics should be kept seperate, because even though they may have the same dimensions and represent similar (or in some cases the same physical quantities), they are used in very different ways by our propagator and numerical integrator, stored in different way (or not stored at all which is indeed a whole other discussion), etc.

I agree with this, but this is precisely the problem I was pointing out. To me, we have a mixture of things, on the one hand we have `State`, on the other we have `{Instant, VectorXd, Frame, ...}` which is essentially a `State`. 

The reason for this mixture as of today is due to Boost, and this is known and accepted. And it's why we have "ugly" tooling like this in `Dynamics`:

![image](https://github.com/open-space-collective/open-space-toolkit-astrodynamics/assets/65550121/68972382-eed0-4c85-a950-991f38fc8649)

(allowing myself to call it "ugly" as I took part in that)

Having said that, we should, imo, try and stop the spread of this limitation outside of the classes that need to interface directly with Boost. Otherwise we will find ourselves coding the exact same for loops, reduction, expansion and sorting logic at the `{VectorXd}` level.

⚠️ IMPORTANT: all in all, I think this should be outside of the scope of this MR. All I was saying is, whether Boost stays or leave, it's blast radius should be limited to a few classes, and we could start putting the work now, asking ourselves every time we operate at `VectorXd` level: "should I make use of `State` here?". 

----

The MR looks good to me 👍

Reply:
ok yeah, great analysis of everything, I agree. I should make a ticket and we can address this in another MR

Reply:
Also @phc1990 to add, I agree with what you said about the `DynamicsContext` and it being indeed a little ugly. I still think "Derivative State" and "Actual State" should be separate classes so that its not mixing too many concepts together, but actually what if we made a base `State` abstract class with the common functionality between "Derivative State" and "Actual State", and then the "Derivative State" and "Actual State" child classes specific functionalities like "computeContribution" for "Derivative State" and perhaps some of the handling of attitude stuff for the "Actual" state (as we currently have in `State`).

Indeed then we could leverage the `StateBuilder` for both the "Actual State" and "Derivative State". I think this would be pretty clean and neat.

Reply:
fully agree 💯